doc/microsite/src/main/tut/distage/distage-testkit.md

```diff
@@ -1,14 +1,518 @@
+returning effects will have arguments provided from the DI graph. These test cases are supported by
+@scaladoc[`in` from DSWordSpecStringWrapper](izumi.distage.testkit.services.scalatest.dstest.DistageAbstractScalatestSpec$$DSWordSpecStringWrapper).
+
+The different effect types fix the `F[_]` argument for this syntax:
+
+- `DistageSpecScalatest`: `F[_]`
+- `DistageBIOSpecScalatest`: `F[Throwable, _]`
+- `DistageBIOEnvSpecScalatest`: `F[Any, Throwable, _]`
+
+For the example application we'll use this to verify the `Score.echoConfig` method.  Additionally,
+the `Config` required is from the `distage` object graph defined in `moduleOverrides` above. By using
+is a function from `Config` the required argument will be injected by `distage-testkit`.
+
+```scala
+package app
+```
+```scala mdoc:to-string
+final class ScoreEffectsTest extends Test {
+  "testkit operations with effects" should {
+
+    "assertions in effects" in { (config: Config) =>
+      for {
+        actual <- Score.echoConfig(config)
+        _ <- assertIO(actual == config)
+      } yield ()
+    }
+
+    "assertions from effects" in { (config: Config) =>
+
+      Score.echoConfig(config) map {
+        actual => assert(actual == config)
+      }
+    }
+  }
+}
+```
+
+```scala mdoc:invisible
+//org.scalatest.shortstacks.nocolor.run(new ScoreEffectsTest)
+```
+
+#### Assertions with Effects with Environments
+
+@scaladoc[The `in` method for `F[_, _, _]` effect types](izumi.distage.testkit.services.scalatest.dstest.DistageAbstractScalatestSpec$$DSWordSpecStringWrapper3)
+supports injection of environments from the object graph. This is in addition to the simple
+assertions and assertions with effects described above.
+
+A test that verifies the bonus service always returns one would be:
+
+```scala
+package app
+```
+```scala mdoc:to-string
+abstract class BonusServiceIsZeroTest extends Test {
+  "BonusService" should {
+    "return one" in {
+      for {
+        bonusService <- ZIO.service[BonusService]
+        currentBonus <- bonusService.queryCurrentBonus
+        _ <- putStrLn(s"currentBonus = ${currentBonus}")
+        _ <- assertIO(currentBonus == 1)
+      } yield ()
+    }
+  }
+}
+```
+
+While this compiles, this test cannot be run without the object graph containing a `BonusService`
+resource.
+For `ZIO[-R, +E, +A]` the `Has` bindings are injected from `ZLayer`, `ZManaged` `ZIO` or any
+`F[_, _, _]: BIOLocal`.
+See @ref[here for details on ZIO Has injection ](basics.md#zio-has-bindings).
+
+Our example application has a dummy and production implementation of the `BonusService`.  For each
+implementation a `ZManaged` is provided. With the `ZManaged` resources added to the object
+graph test cases can inject `Has[BonusService]`.
+
+For demonstration of [reuse and memoization](#resource-reuse-memoization) the bonus value will be
+equal to the number of times the resource was acquired.
+
+```scala
+package app
```

Comment:
Are you planning on making the package significant in some way (e.g. plugins?) For plugins we may only need the plugin to be defined in a non-default package. (I think mdoc will not allow defining something in a package, but you may add any code you need to `doc/microsite/src/main/scala` and reference it in the docs, e.g. plugins)

Reply:
Right. Demonstrating plugin functionality is the goal. Specifically how the the default test config included the current package's plugins.
RE: mdoc not supporting defining something in a package. I encountered that as well. That's why the `package` statements are in plain scala blocks. A bit odd no? Still, somewhat captures the importance of packages.
 
You might have seen the note below related to running a test with the fake package: https://github.com/7mind/izumi/pull/1145/commits/32965b436e016edd4e9d540ece8aac60e83dabb3#diff-9cc6af0950016de37c81fb0f5609a6e6R403

I'll see about cleaning that up using `src/main/scala`. Was not aware of that feature. Thanks!

Reply:
@coreyoconnor Hi, I've added a modifier to mdoc to allow writing a (fake) package statement in the same block - hope that helps! https://github.com/dogheadbone/izumi/pull/1